src/Extend/Post.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * For detailed copyright and license information, please view the
+ * LICENSE file that was distributed with this source code.
+ */
+
+namespace Flarum\Extend;
+
+use Flarum\Extension\Extension;
+use Flarum\Post\Post as ActualPost;
+use Illuminate\Contracts\Container\Container;
+
+class Post implements ExtenderInterface
+{
+    /**
+     * Register a new post type. This is generally done for custom 'event posts',
+     * such as those that appear when a discussion is renamed.
+     *
+     * @param string $postType: The ::class attribute of the custom Post type that is being added.
+     */
+    public function type($postType)
+    {
+        ActualPost::setModel($postType::$type, $postType);
+
+        return $this;
+    }
+
+    public function extend(Container $container, Extension $extension = null)
+    {
+        // Nothing happens here, type does all the work.
```

Comment:
There's probably not much of a difference, but wouldn't it still be better to store posts in an array and call `setModel` from here?

I believe this might delay the resolving of the classname until extensions actually boot? Not sure how that would be useful but resolving as late as possible is always nice.